src/main/java/uk/gov/admin/DataFileReader.java

```diff
@@ -40,6 +47,23 @@ public DataFileReader(String datafile, String type) throws URISyntaxException {
         }
     }
 
+    private Iterator<String> readYamlFiles() throws IOException {
+        Iterator<Path> filesIterator = Files.newDirectoryStream(Paths.get(dataSourceURI)).iterator();
+        return Stream.of(Iterators.toArray(filesIterator, Path.class))
+                .filter(path -> path.toString().endsWith(".yaml"))
+                .map(this::convertToJsonEntry)
+                .collect(Collectors.toList())
```

Comment:
you don't need this line -- you can convert the Stream directly to an Iterator with BaseStream.iterator().